test/tools/d_do_test.d

```diff
@@ -532,20 +528,35 @@ int main(string[] args)
 
 int tryMain(string[] args)
 {
-    if (args.length != 4)
+    if (args.length != 2)
     {
         usage();
         return 1;
     }
 
-    string input_dir      = args[1];
-    string test_name      = args[2];
-    string test_extension = args[3];
+    auto test_file = args[1];
+    auto result = matchFirst(test_file, `^([^/^\\]+)[/\\]([^\.]+)\.(.+)$`);
+    if (result.empty)
+    {
+        writefln("Error: test file '%s' did not match regex", test_file);
+        return 1;
+    }
+    string input_dir = result[1];
+    string test_name = result[2];
+    string test_extension = result[3];
 
     EnvData envData;
-    envData.all_args      = environment.get("ARGS");
-    envData.results_dir   = environment.get("RESULTS_DIR");
     envData.sep           = envGetRequired ("SEP");
+
+    if (test_extension == "sh")
+    {
+        auto process = spawnProcess(["./tools/sh_do_test.sh".replace("/", envData.sep),
+            input_dir, test_name]);
+        return process.wait();
```

Comment:
Why not `std.process.execute` if you wait anyhow?

Reply:
not capturing stdout/stderr